### PR TITLE
llama3: move dl to numpy & jit more

### PR DESCRIPTION
**Pull request by wozeparrot**
image

### CODE REVIEW

File: examples/mlperf/model_train.py [hunk 1]
```diff
@@ -1401,7 +1402,7 @@ def minibatch(tokens:Tensor):
     loss.backward()
     assert all(p.grad is g for p,g in zip(optim.params, grads))
     Tensor.realize(loss, *grads)
-    return loss
+    return loss.flatten().float().to("CPU")
```
Comment (chenyuxyz) [inline examples/mlperf/model_train.py:1405]: is scheduler smart enough to move float after the to to save bandwidth?

Comment (wozeparrot) [reply examples/mlperf/model_train.py:1405]: no this does the .float on the gpu. i tried putting the .float after but the nothing really changes step time wise.

Comment (geohot) [reply examples/mlperf/model_train.py:1405]: i mean...it's 4 bytes vs 2 bytes
